src/idempotent.ts

```diff
@@ -1,34 +1,68 @@
-export interface IdempotentStore {
-  get<T = any>(request: string): Promise<{ found: boolean; result: T }>;
-  set<T>(request: string, response: T): Promise<boolean>;
-  lock(request: string): Promise<boolean>;
-  release(request: string): Promise<boolean>;
-}
+import { FibonacciStrategy, Backoff } from 'backoff';
 
-export class LockError extends Error {
-  constructor() {
-    super('Failed to lock the request id');
-  }
+export interface IdempotentStore {
+  get(id: string): Promise<any | null>;
+  set(
+    id: string,
+    params:
+      | {
+          status: 'STARTED';
+        }
+      | {
+          status: 'DONE';
+          result: any;
+        }
+  ): Promise<boolean>;
 }
 
 export class Idempotent {
-  constructor(private store: IdempotentStore) {}
+  constructor(private readonly store: IdempotentStore) {}
 
   async execute<T = any>(fn: () => Promise<T>, request: string): Promise<T> {
-    const lock = await this.store.lock(request);
-    if (!lock) {
-      throw new LockError();
-    }
-    const response = await this.store.get(request);
-    if (response && response.found) {
```

Comment:
I did this because the result can be falsy,. like a handler returning nothing but the request succeeded and cached,.

Reply:
The subject for this comment is not clear for me. Can you please explain more?

Reply:
okay, I think I understand your point now, so you want to handle cases where the handler returns a falsy?

Reply:
yeah,. there some cases whereby the `fn` will return a falsy.